### PR TITLE
fix(saved-trip-editor): Route back button to saved trips if editing

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**

Fixes a bug where, if a user accessed the "edit trip" page through a link in their email, the "back" button would send them to a broken Auth0 url. 

Now if a user is editing a trip, it will always take you back to saved trips. 

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/components/user/back-link.js [hunk 1]
```diff
@@ -12,14 +12,18 @@ const StyledButton = styled(Button)`
   padding: 0;
 `
 
+export const backButtonContent = (
```
Comment (binh-dam-ibigroup) [inline lib/components/user/back-link.js:15]: Maybe this should be a component.

File: lib/components/user/monitored-trip/saved-trip-editor.tsx [hunk 2]
```diff
@@ -52,7 +54,13 @@ const SavedTripEditor = (props: Props): JSX.Element => {
     return (
       <>
         <PageTitle title={title} />
-        <BackLink />
+        {/* If creating, back button should return to trip planner. If not, it should return
+        to saved trips, just in case the user accessed this page through their email. */}
+        {isCreating ? (
+          <BackLink />
```
Comment (binh-dam-ibigroup) [inline lib/components/user/monitored-trip/saved-trip-editor.tsx:60]: If we are doing links, how about we place a `<Link to="/">` for when `isCreating` is true? (The link will preserve query params in the URL)

**Review by binh-dam-ibigroup — COMMENTED** [review]
What do you think about using a link to "/" when creating trips?

File: lib/components/user/monitored-trip/saved-trip-editor.tsx (hunk 2 above)
Comment (amy-corson-ibigroup) [reply lib/components/user/monitored-trip/saved-trip-editor.tsx]: I think you're right, added in [da8783a](https://github.com/opentripplanner/otp-react-redux/pull/1255/commits/da8783aa0e80dd3ab09e6b4200fe23a61ac7d8b0)

File: lib/components/user/back-link.js (hunk 1 above)
Comment (amy-corson-ibigroup) [reply lib/components/user/back-link.js]: Good call! Fixed in [127475c](https://github.com/opentripplanner/otp-react-redux/pull/1255/commits/127475c1cf79ef983aba3a5cf13f0cf6a255eb00)

**Review by binh-dam-ibigroup — APPROVED** [review]
Sweet!

**Comment by amy-corson-ibigroup** [comment]
@miles-grant-ibigroup @binh-dam-ibigroup  if this is looking okay after Miles reviews, and you guys wouldn't mind merging while I'm out next week, that would be amazing!

**Review by miles-grant-ibigroup — APPROVED** [review]
I'm tempted to rename `Content` to `Text` or similar but it's not worth blocking this whole thing

**Pull request merged**